proxy/httpproxy/reverse.go

```diff
@@ -110,15 +109,16 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	var requestClosed int32
 	completeCh := make(chan bool, 1)
 	closeNotifier, ok := rw.(http.CloseNotifier)
-	cancel := httputil.RequestCanceler(proxyreq)
+	ctx, cancel := context.WithCancel(context.Background())
+	proxyreq = proxyreq.WithContext(ctx)
+	defer cancel()
 	if ok {
 		closeCh := closeNotifier.CloseNotify()
 		go func() {
 			select {
 			case <-closeCh:
 				atomic.StoreInt32(&requestClosed, 1)
 				plog.Printf("client %v closed request prematurely", clientreq.RemoteAddr)
-				cancel()
```

Comment:
@xiang90 its not required there. It will still work fine. The reason being the code is written in such a way that on either of these select triggered, the method will exit and also cancel will get called. The defer func after the select causes it to end i think.